warp/src/js_exports/raygun.rs

```diff
@@ -301,6 +304,137 @@ impl RayGunBox {
     }
 }
 
+/// impl RayGunGroup trait
+#[wasm_bindgen]
+impl RayGunBox {
+    //Update conversation name Note: This will only update the group conversation name
+    pub async fn update_conversation_name(
+        &mut self,
+        conversation_id: String,
+        name: String,
+    ) -> Result<(), JsError> {
+        self.inner
+            .update_conversation_name(Uuid::from_str(&conversation_id).unwrap(), &name)
+            .await
+            .map_err(|e| e.into())
+    }
+
+    /// Add a recipient to the conversation
+    pub async fn add_recipient(
+        &mut self,
+        conversation_id: String,
+        recipient: String,
+    ) -> Result<(), JsError> {
+        self.inner
+            .add_recipient(
+                Uuid::from_str(&conversation_id).unwrap(),
+                &DID::from_str(&recipient).unwrap_or_default(),
+            )
+            .await
+            .map_err(|e| e.into())
+    }
+
+    /// Remove a recipient from the conversation
+    pub async fn remove_recipient(
+        &mut self,
+        conversation_id: String,
+        recipient: String,
+    ) -> Result<(), JsError> {
+        self.inner
+            .remove_recipient(
+                Uuid::from_str(&conversation_id).unwrap(),
+                &DID::from_str(&recipient).unwrap_or_default(),
+            )
+            .await
+            .map_err(|e| e.into())
+    }
+}
+
+/// impl RayGunAttachment trait
+#[wasm_bindgen]
+impl RayGunBox {
+    /// Send files to a conversation.
+    /// If no files is provided in the array, it will throw an error
+    pub async fn attach(
+        &mut self,
+        conversation_id: String,
+        message_id: Option<String>,
+        locations: Vec<JsValue>,
+        message: Vec<String>,
+    ) -> Result<AttachmentResult, JsError> {
+        self.inner
+            .attach(
+                Uuid::from_str(&conversation_id).unwrap(),
+                message_id.map(|s| Uuid::from_str(&s).unwrap()),
+                locations
+                    .iter()
+                    .map(|v| serde_wasm_bindgen::from_value(v.clone()).unwrap())
+                    .collect(),
+                message,
+            )
+            .await
+            .map_err(|e| e.into())
+            .map(|(id, ok)| AttachmentResult {
+                file: id.to_string(),
+                stream: AsyncIterator::new(Box::pin(
+                    ok.map(|s| serde_wasm_bindgen::to_value(&AttachmentKind::from(s)).unwrap()),
+                )),
+            })
+    }
```

Comment:
As a note, you would not be able to use `Location::Path` since that relies on the filesystem. I would suggest uploading to constellation first and use `Location::Constellation` for the time being until `Location` supports piping a stream for a specific file.

Reply:
Since we cant access the filesystem in wasm, we would need to rely on other means since we cannot read a file using `Location::Path` from the browser, which in this case would use contellation or passing a valid stream of bytes that we can poll and pass into constellation. A workaround for now would be to use `Location::Constellation`, which would require the frontend to upload to constellation first before attaching it in a message since it would already be available there.

Reply:
i added the version that now uses a stream

Reply:
Lets focus on the streams in a separate PR as we can use reuse `RayGun::attach` by adding streams to `Location` instead of introducing a new object and new function

Reply:
i added it as something new cause i think streams are not cloneable? which location requires

Reply:
We can have something inclusive to `js_export` so it could probably either be split (similar to what is done for `MultiPass::get_identity`) or have a separate function exclusive to js export